src/adaptors/mod.rs

```diff
@@ -5,8 +5,10 @@
 //! except according to those terms.
 
 mod coalesce;
+mod map;
 mod multi_product;
 pub use self::coalesce::*;
+pub use self::map::*;
```

Comment:
Does `MapSpecialCase` need to be part of itertools's public API? Instead of glob importing everything from `map`, perhaps just import the exact items we want to be part of our public API.

Reply:
Good idea.